templates/repo/settings/secrets.tmpl

```diff
@@ -0,0 +1,69 @@
+<div class="ui container">
+		<h4 class="ui top attached header">
+			{{.locale.Tr "repo.settings.secrets"}}
+			<div class="ui right">
+				<div class="ui primary tiny show-panel button" data-panel="#add-secret-panel">{{.locale.Tr "repo.settings.add_secret"}}</div>
+			</div>
+		</h4>
+		<div class="ui attached segment">
+			<div class="{{if not .HasError}}hide{{end}} mb-4" id="add-secret-panel">
+				<form class="ui form" action="{{.Link}}?act=secret" method="post">
+					{{.CsrfTokenHtml}}
+					<div class="field">
+						{{.locale.Tr "repo.settings.secret_desc"}}
+					</div>
+					<div class="field {{if .Err_Title}}error{{end}}">
+						<label for="title">{{.locale.Tr "repo.settings.secret_key"}}</label>
+						<input id="ssh-key-title" name="title" value="{{.title}}" autofocus required>
+					</div>
+					<div class="field {{if .Err_Content}}error{{end}}">
+						<label for="content">{{.locale.Tr "repo.settings.secret_content"}}</label>
+						<textarea id="ssh-key-content" name="content" placeholder="{{.locale.Tr "repo.settings.secret_value_content_placeholder"}}" required>{{.content}}</textarea>
+					</div>
+					<div class="field">
+						<div class="ui checkbox {{if .Err_IsWritable}}error{{end}}">
+							<input id="pull_request_read" name="pull_request_read" class="hidden" type="checkbox" value="1">
+							<label for="is_writable">
+								{{.locale.Tr "repo.settings.pull_request_read"}}
+							</label>
+							<small style="padding-left: 26px;">{{$.locale.Tr "repo.settings.pull_request_read_info" | Str2html}}</small>
+						</div>
+					</div>
+					<button class="ui green button">
+						{{.locale.Tr "repo.settings.add_secret"}}
+					</button>
+					<button class="ui hide-panel button" data-panel="#add-secret-panel">
+						{{.locale.Tr "cancel"}}
+					</button>
+				</form>
+			</div>
+			{{if .Secrets}}
+				<div class="ui key list">
+					{{range .Secrets}}
+						<div class="item">
+							<div class="right floated content">
+								<button class="ui red tiny button delete-button" data-url="{{$.Link}}/delete?act=secret" data-id="{{.ID}}">
+									{{$.locale.Tr "settings.delete_key"}}
+								</button>
+							</div>
+							<div class="left floated content">
+								<i class="tooltip">{{svg "octicon-key" 32}}</i>
+							</div>
+							<div class="content">
+								<strong>{{.Name}}</strong>
+								<div class="print meta">
+									{{Shadow .Data}}
+								</div>
+								<div class="activity meta">
+									<i>{{$.locale.Tr "settings.add_on"}} <span>{{.CreatedUnix.FormatShort}}</span>
+									{{if .PullRequest}} —  {{svg "octicon-info"}} - <span> {{$.locale.Tr "repo.settings.pull_request_read_hint"}} </span>{{end}}</i>
```

Comment:
That looks hacky. Is there no better approach for this?
Especially: Why are there two different types of hyphens?